### ISSUE TITLE
Internal Server Error on Endpoint Status creation without date field (version 2.17.0)

### ISSUE THREAD

**Issue by rgorosito** [DefectDojo/django-DefectDojo#7336]
I've upgraded from DD 2.15.1 to 2.17.0 and some process stopped working with endpoint status creation errors.

I was able to reproduce it with a clean install using docker-compose in local environment.

**Steps to reproduce**
1. git clone https://github.com/DefectDojo/django-DefectDojo
2. cd django-DefectDojo
3. git checkout 2.17.0
4. ./dc-build.sh
5. ./dc-up.sh postgres-rabbitmq
6. create a product, engagement, test, endpoint
7. go to "API v2 OpenAPI3 Docs"
8. go to "endpoint_status" -> "POST" -> "try it out"
9. replace "request body" with: '''{
  "endpoint": 1,
  "finding": 1
}'''
10. execute

you will obtain an internal server error in server response:
```
{
  "message": "Internal server error, check logs for details"
}
```

**Console error** 
```
uwsgi_1         | [30/Dec/2022 12:49:02] ERROR [dojo.api_v2.exception_handler:32] Expected a `date`, but got a `datetime`. Refusing to coerce, as this may mean losing timezone information. Use a custom read-only field and deal with timezone issues explicitly.
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
uwsgi_1         |     response = handler(request, *args, **kwargs)
uwsgi_1         |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/mixins.py", line 20, in create
uwsgi_1         |     headers = self.get_success_headers(serializer.data)
uwsgi_1         |                                        ^^^^^^^^^^^^^^^
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/serializers.py", line 555, in data
uwsgi_1         |     ret = super().data
uwsgi_1         |           ^^^^^^^^^^^^
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/serializers.py", line 253, in data
uwsgi_1         |     self._data = self.to_representation(self.instance)
uwsgi_1         |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/serializers.py", line 522, in to_representation
uwsgi_1         |     ret[field.field_name] = field.to_representation(attribute)
uwsgi_1         |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
uwsgi_1         |   File "/usr/local/lib/python3.11/site-packages/rest_framework/fields.py", line 1252, in to_representation
uwsgi_1         |     assert not isinstance(value, datetime.datetime), (
uwsgi_1         | AssertionError: Expected a `date`, but got a `datetime`. Refusing to coerce, as this may mean losing timezone information. Use a custom read-only field and deal with timezone issues explicitly.
uwsgi_1         | [30/Dec/2022 12:49:02] ERROR [django.request:224] Internal Server Error: /api/v2/endpoint_status/
```

**Aditional information**
* Endpoint status is created despite the error message
* If field date is included in the request, it doesn't fail.
Example:

Request:
```
{
  "date": "2022-12-30",
  "endpoint": 1482,
  "finding": 140324
}
```

Response:
```
{
  "id": 285529,
  "date": "2022-12-30",
  "last_modified": "2022-12-30T12:20:10.395190Z",
  "mitigated": false,
  "mitigated_time": null,
  "false_positive": false,
  "out_of_scope": false,
  "risk_accepted": false,
  "mitigated_by": null,
  "endpoint": 1482,
  "finding": 140324
}
```

**Comment by rgorosito** [comment]
endpoint_status is created, but it is not returning the json with the id. It's returning this Internal Server Error

**Comment by rgorosito** [comment]
Update: If field date is included in the request, it doesn't fail.
Example:

Request:
```
{
  "date": "2022-12-30",
  "endpoint": 1482,
  "finding": 140324
}
```

Response:
```
{
  "id": 285529,
  "date": "2022-12-30",
  "last_modified": "2022-12-30T12:20:10.395190Z",
  "mitigated": false,
  "mitigated_time": null,
  "false_positive": false,
  "out_of_scope": false,
  "risk_accepted": false,
  "mitigated_by": null,
  "endpoint": 1482,
  "finding": 140324
}
```

**Comment by kiblik** [comment]
I prepared fix: #7395 (including the test)

@rgorosito, are you able to test it as well with this fix, please?

**Comment by rgorosito** [comment]
Sorry for the delay. Thanks @kiblik . I confirm that it's fixed now
